### PR TITLE
feat: bump keyring controller 7.5.0

### LINKED ISSUE #8180: Bump `@metamask/keyring-controller` from v6 to v7
### What is this about?

Bump `@metamask/keyring-controller` from v6 to v7

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [X] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by gantunesr**
## **Description**

This PR bumps the `@metamask/keyring-controller` version from `6.0.0` to `7.5.0`. These are the relevant changes to take into consideration during review,

- **BREAKING**: Remove `keyringTypes` property from the KeyringController state ([#1441](https://github.com/MetaMask/core/pull/1441)) 
- **BREAKING**: Constructor `KeyringControllerOptions` type changed ([#1441](https://github.com/MetaMask/core/pull/1441))
  - The `KeyringControllerOptions.state` accepted type is now `{ vault?: string }`
  - The `KeyringControllerOptions.keyringBuilders` type is now `{ (): Keyring<Json>; type: string }[]`
- **BREAKING**: The `address` type accepted by the `removeAccount` method is now `Hex` ([#1441](https://github.com/MetaMask/core/pull/1441))
- **BREAKING**: The `signTypedMessage` method now returns a `Promise<string>` ([#1441](https://github.com/MetaMask/core/pull/1441))   
- **BREAKING**: The `signTransaction` method now requires a `TypedTransaction` from `@ethereumjs/tx@^4` for the `transaction` argument, and returns a `Promise<TxData>` ([#1441](https://github.com/MetaMask/core/pull/1441))
- **BREAKING:** Rename `Keyring` type to `KeyringObject` ([#1441](https://github.com/MetaMask/core/pull/1441))
- **BREAKING:** `addNewAccount` now throws if address of new account is not a hex string ([#1441](https://github.com/MetaMask/core/pull/1441))
- **BREAKING:** `exportSeedPhrase` now throws if first keyring does not have a mnemonic ([#1441](https://github.com/MetaMask/core/pull/1441))
- **BREAKING:** `verifySeedPhrase` now throws if HD keyring does not have a mnemonic ([#1441](https://github.com/MetaMask/core/pull/1441))
 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/8180

## **Manual testing steps**

Use cases or flows to verify,

1. Onboarding
2. Import SRP
3. Import private key
4. Reveal SRP
5. Reveal private key
6. Add new account
7. Connect QR wallet
8. Signed type messages
9. Remove imported account
10. Lock and unlock wallet

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/Engine.ts [hunk 1]
```diff
@@ -547,7 +547,11 @@ class Engine {
         ],
         allowedActions: ['KeyringController:getState'],
       }),
-      state: initialKeyringState || initialState.KeyringController,
+      state: {
+        vault:
```
Comment (owencraston) [inline app/core/Engine.ts:551]: why not pass the entire state object?

**Comment by owencraston** [comment]
Confirmed the engine is still being backed up
Screenshot 2023-12-21 at 12 42 30 PM

**Review by owencraston — DISMISSED** [review]
I was able to import and create an account. The balances were correct and I was able to send a transaction.
Simulator Screenshot - iPhone 15 Pro - 2023-12-21 at 12 43 20

File: app/core/Ledger/Ledger.ts [hunk 2]
```diff
@@ -140,8 +111,7 @@ export const unlockLedgerDefaultAccount = async (): Promise<{
       keyringController.setSelectedAddress(address);
```
Comment (Gudahtt) [inline app/core/Ledger/Ledger.ts:111]: Hmm, this should be private. Curious that this didn't result in a type error. I'm guessing that `keyringController` doesn't have the correct type here.

File: app/core/Engine.ts (hunk 1 above)
Comment (gantunesr) [reply app/core/Engine.ts]: True, solved in [840658e](https://github.com/MetaMask/metamask-mobile/pull/8035/commits/840658e506da94bb48791070098394058e846149)

File: app/core/Ledger/Ledger.ts (hunk 2 above)
Comment (gantunesr) [reply app/core/Ledger/Ledger.ts]: Right, I missed that. Update to use the `PreferencesController` instead

Comment (gantunesr) [reply app/core/Ledger/Ledger.ts]: [315fe04](https://github.com/MetaMask/metamask-mobile/pull/8035/commits/315fe0408a504860d10c642506910e9bfb1e50d8)

**Review by owencraston — APPROVED** [review]

**Comment by vivek-consensys** [comment]
All tests have passed, all manual test cases were carried out using both OS Android 13 and iOS 17.1.1.
Bitrise QA Build used to test:- https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/1b46d210-0487-4f99-a638-09c33c6d4706

**Comment by plasmacorral** [comment]
Tested commit 28f05a6a2c26e805130a8b8f9aa4cc340e48a486 with both Samsung A515f on Android 12 with fingerprint and iPhone 13 mini on iOS 15.6.2 with faceID.

Tested the following with no issues:
- Upgrade testing from V7.12.2 build 1224
  - all accounts present
  - custom account names preserved
- Onboarding 
  - create new SRP
  - import SRP
- Delete wallet from settings
- Reset wallet from login
- Import private key
- Reveal SRP
- Reveal private key
- Add new HD account
- Edit account name
- HD and imported account
  - Testnet send tx
  - Personal sign
  - Typed data V1
  - Typed data V3
  - Typed data V4
  - Sign permit
  - SIWE
- Lock and unlock wallet
  - Manually lock from settings, then unlock
  - app autolock at 30s when in background with display remaining active, then open and unlock
  - Kill app and open then unlock
  - Set device display lock setting to 1 min with default autolock 30s and let screen timeout, then unlock device and unlock wallet

**Pull request merged**